fili-core/src/main/java/com/yahoo/bard/webservice/table/availability/ConcreteAvailability.java

```diff
@@ -105,4 +105,8 @@ public SimplifiedIntervalList getAvailableIntervals(DataSourceConstraint constra
     protected Map<String, List<Interval>> getLatestAvailableIntervalsByTable() {
         return metadataService.getAvailableIntervalsByTable(name);
     }
+
+    public Set<String> getColumnNames() {
```

Comment:
The loss is that `PermissiveAvailability` wouldn't be able to get at the column names, which it needs it. So, I think making it `protected` is the best bet for it, rather than dropping it.